attachments/forms.py

```diff
@@ -56,16 +56,24 @@ def __init__(self, *args, **kwargs):
         is_upload = isinstance(instance, Upload)
         is_attachment = isinstance(instance, Attachment)
         form_data = self.get_form_data_from_session_data(instance.session.data) if is_upload else {}
+        
+        if not backend_data:
+            backend_data = instance.session.backend_data
```

Comment:
Only Uploads have an associated session, so this will fail for Attachment objects.